packages/sanity/src/core/hooks/useEditState.ts

```diff
@@ -25,5 +26,10 @@ export function useEditState(
     }
 
     return documentStore.pair.editState(publishedDocId, docTypeName)
-  }, [documentStore.pair, publishedDocId, docTypeName, priority]) as EditStateFor
+  }, [docTypeName, documentStore.pair, priority, publishedDocId])
+  /**
+   * We know that since the observable has a startWith operator, it will always emit a value
+   * and that's why the non-null assertion is used here
+   */
+  return useObservable(observable)!
```

Comment:
I tried to set the `initialValue` here instead of the non-null assertion. Unfortunately some of the initial state is non-trivial to work out: https://github.com/sanity-io/sanity/blob/62581ba91e96de16b7f23e4afdf1b92e9f234602/packages/sanity/src/core/store/_legacy/document/document-pair/editState.ts#L68

Reply:
We should be good with a non-null assertion since we know these has sync emissions (unfortunately there's no way to express this constraint with TS).